### PR TITLE
Store tables without yang

### PR DESCRIPTION
**Pull request by praveen-li**
**- What I did**

Store Tables without YANG models seperately for user warning.

**- How I did it**

Changes:
    1.) Store Tables with YANG models seperately, so that a warning to user
    can be issued with enough information.
    2.) print the tables without YANG models properly in cropConfigDB().

**- How to verify it**

    Changes:
    1.) Added config for extra tables in yangTest.json.
    2.) Added new test case to find tables without yang models.
    3.) Move common part in a helper function.

```
pchaudha@be1e6e58be56:/sonic$ make -f slave.mk target/python-wheels/sonic_yang_mgmt-1.0-py2-none-any.whl
SONiC Build System

Build Configuration
"CONFIGURED_PLATFORM"             : "vs"
"CONFIGURED_ARCH"                 : "amd64"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "1"
"SONIC_CONFIG_MAKE_JOBS"          : "24"
"SONIC_USE_DOCKER_BUILDKIT"       : ""
"USERNAME"                        : "admin"
"PASSWORD"                        : "YourPaSsWoRd"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_SYSTEM_TELEMETRY"         : "y"
"ENABLE_RESTAPI"                  : "n"
"ENABLE_ZTP"                      : ""
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : ""
"BLDENV"                          : ""
"VS_PREPARE_MEM"                  : "yes"
"ENABLE_SFLOW"                    : "y"
"ENABLE_NAT"                      : "y"

pchaudha@be1e6e58be56:/sonic$ cat target/python-wheels/sonic_yang_mgmt-1.0-py2-none-any.whl.log

tests/test_sonic_yang_mgmt.py .
tests/libyang-python-tests/test_sonic_yang.py ........................

========================== 25 passed in 0.27 seconds ===========================
```

Test on 06/29/2020
```
pchaudha@server05:/home/pchaudha/srcCode/dpb_repo$ ls -l target/python-wheels/sonic_yang_mgmt-1.0-py2-none-any.whl
-rw-r--r-- 1 pchaudha pchaudha 19385 Jun 29 23:04 target/python-wheels/sonic_yang_mgmt-1.0-py2-none-any.whl
pchaudha@server05:/home/pchaudha/srcCode/dpb_repo$ date
Mon Jun 29 23:09:25 PDT 2020
pchaudha@server05:/home/pchaudha/srcCode/dpb_repo$
```
**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 1]
```diff
@@ -117,16 +118,22 @@ def get_module_TLC_container(self, table):
...
-                raise(Exception("No Yang Model Exist for {}".format(table)))
+            # store in tablesWithOutYang
+            self.tablesWithOutYang[table] = self.jIn[table]
+            del self.jIn[table]
+
+    if len(self.tablesWithOutYang):
+        print("Note: Below table(s) have no YANG models:")
+        for table in self.tablesWithOutYang.keys():
+            print(unicode(table), end=", ")
+        print()
```
Comment (zhenggen-xu) [inline src/sonic-yang-mgmt/sonic_yang_ext.py]: what is this print() for?

File: src/sonic-yang-mgmt/tests/libyang-python-tests/test_sonic_yang.py [hunk 2]
```diff
@@ -266,20 +266,38 @@ def test_get_leafref_type_schema(self, yang_s, data):
...
-        jIn = self.readIjsonInput(yang_test_file, 'SAMPLE_CONFIG_DB_JSON')
-        # load yang models
-        syc = sy.sonic_yang(yang_dir)
-
-        syc.loadYangModel()
+        jIn = self.readIjsonInput(test_file, 'SAMPLE_CONFIG_DB_JSON')
 
         syc.load_data(json.loads(jIn))
 
+        # TODO: Make sure no extra table is loaded
```
Comment (zhenggen-xu) [inline src/sonic-yang-mgmt/tests/libyang-python-tests/test_sonic_yang.py:296]: Can you elaborate the missing function here?

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 3]
```diff
@@ -117,16 +118,22 @@ def get_module_TLC_container(self, table):
...
-            if allowExtraTables:
-                del self.jIn[table]
-            else:
-                raise(Exception("No Yang Model Exist for {}".format(table)))
+            # store in tablesWithOutYang
+            self.tablesWithOutYang[table] = self.jIn[table]
+            del self.jIn[table]
+
+    if len(self.tablesWithOutYang):
+        print("Note: Below table(s) have no YANG models:")
```
Comment (zhenggen-xu) [inline src/sonic-yang-mgmt/sonic_yang_ext.py]: We should log to syslog too.

File: src/sonic-yang-mgmt/sonic_yang_ext.py (hunk 1 above)
Comment (praveen-li) [reply src/sonic-yang-mgmt/sonic_yang_ext.py]: print(unicode(table), end=", ") prints all extra tables in one line, then print() is just used to change line.

Comment (praveen-li) [reply src/sonic-yang-mgmt/sonic_yang_ext.py]: Same as https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-mgmt/sonic_yang_ext.py

File: src/sonic-yang-mgmt/tests/libyang-python-tests/test_sonic_yang.py (hunk 2 above)
Comment (praveen-li) [reply src/sonic-yang-mgmt/tests/libyang-python-tests/test_sonic_yang.py:296]: Right now, In test file (i.e. sample config) we have extra tables, for which yang models are not merged yet: 
https://github.com/Azure/sonic-buildimage/pull/4442.

This line is to remind, once all yang models are merged, I can even test here that no extra tables exist in test config.

File: src/sonic-yang-mgmt/sonic_yang_ext.py (hunk 3 above)
Comment (praveen-li) [reply src/sonic-yang-mgmt/sonic_yang_ext.py]: Done

**Review by vasant17 — APPROVED** [review]

**Pull request merged**